File: flutter/lib/screens/projects_screen.dart
// Projects list — large title, portfolio summary (teal AVG ring + status counts),
// single-select filter chips with live filtering, and project cards (mini ring +
// name + "{location} · {area} m² · {scans} scans" + status badge + chevron).
// Tapping a card pushes the Report detail. Ports pwa ProjectsList.
import 'package:flutter/material.dart';
import '../data.dart';
import '../theme.dart';
import '../widgets/ring.dart';
import '../widgets/status_badge.dart';
import '../widgets/ui.dart';
import 'report_detail_screen.dart';

class ProjectsScreen extends StatefulWidget {
  const ProjectsScreen({super.key});

  @override
  State<ProjectsScreen> createState() => _ProjectsScreenState();
}

class _ProjectsScreenState extends State<ProjectsScreen> {
  String _filter = 'All';

  List<Project> get _list => kProjects.where((p) {
    switch (_filter) {
      case 'On site':
        return p.status != 'Complete';
      case 'Needs review':
        return p.status == 'Needs Review';
      case 'Complete':
        return p.status == 'Complete';
      default:
        return true;
    }
  }).toList();

  @override
  Widget build(BuildContext context) {
    final avg = portfolioAvg(kProjects);
    final onTrack = kProjects.where((p) => p.status == 'On Track').length;
    final review = kProjects.where((p) => p.status == 'Needs Review').length;
    final done = kProjects.where((p) => p.status == 'Complete').length;
    final list = _list;

    return SafeArea(
      bottom: false,
      child: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(
            child: ScreenHeader(
              title: 'Projects',
              sub: '${kProjects.length} projects · Scotland',
            ),
          ),

          // portfolio summary — the screen's teal hero
          SliverToBoxAdapter(
            child: Padding(
              padding: const EdgeInsets.fromLTRB(20, 2, 20, 8),
              child: AppCard(
                child: Row(
                  children: [
                    Ring(
                      value: avg.toDouble(),
                      size: 66,
                      stroke: 6,
                      accent: true,
                      label: 'AVG',
                    ),
                    const SizedBox(width: 16),
                    Expanded(
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          _countCell('On track', onTrack, T.accent),
                          _countCell('Review', review, T.warning),
                          _countCell('Complete', done, T.muted),
                        ],
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ),

          SliverToBoxAdapter(
            child: Padding(
              padding: const EdgeInsets.only(top: 4, bottom: 10),
              child: Chips(
                items: const ['All', 'On site', 'Needs review', 'Complete'],
                active: _filter,
                onPick: (v) => setState(() => _filter = v),
              ),
            ),
          ),

          if (list.isEmpty)
            const SliverToBoxAdapter(
              child: Padding(
                padding: EdgeInsets.symmetric(vertical: 40),
                child: Center(
                  child: Text(
                    'No projects in this filter.',
                    style: TextStyle(color: T.faint, fontSize: 14),
                  ),
                ),
              ),
            )
          else
            SliverPadding(
              padding: const EdgeInsets.fromLTRB(20, 4, 20, 16),
              sliver: SliverList.separated(
                itemCount: list.length,
                separatorBuilder: (_, _) => const SizedBox(height: 10),
                itemBuilder: (context, i) => _ProjectCard(project: list[i]),
              ),
            ),
        ],
      ),
    );
  }

  Widget _countCell(String label, int n, Color color) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      mainAxisSize: MainAxisSize.min,
      children: [
        Text(
          '$n',
          style: TextStyle(
            fontSize: 22,
            fontWeight: FontWeight.w800,
            color: color,
          ),
        ),
        Text(
          label,
          style: const TextStyle(
            fontSize: 11.5,
            fontWeight: FontWeight.w600,
            color: T.muted,
          ),
        ),
      ],
    );
  }
}

class _ProjectCard extends StatelessWidget {
  const _ProjectCard({required this.project});
  final Project project;

  @override
  Widget build(BuildContext context) {
    final p = project;
    return AppCard(
      padding: const EdgeInsets.all(13),
      onTap: () => Navigator.of(context).push(
        MaterialPageRoute(builder: (_) => ReportDetailScreen(project: p)),
      ),
      child: Row(
        children: [
          Ring(value: p.pct.toDouble(), size: 54),
          const SizedBox(width: 14),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisSize: MainAxisSize.min,
              children: [
                Text(
                  p.name,
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  style: const TextStyle(
                    fontSize: 16,
                    fontWeight: FontWeight.w700,
                    letterSpacing: -0.3,
                  ),
                ),
                const SizedBox(height: 3),
                Text(
                  '${p.location} · ${p.area} m² · ${p.scans} scans',
                  style: const TextStyle(fontSize: 12.5, color: T.muted),
                ),
                const SizedBox(height: 8),
                StatusBadge(status: p.status, small: true),
              ],
            ),
          ),
          const Icon(Icons.chevron_right, size: 20, color: Color(0x38FFFFFF)),
        ],
      ),
    );
  }
}

File: ios-native/Sources/ProjectsScreen.swift
// Projects list — large title, portfolio summary (teal ring + status counts),
// single-select filter chips, and project cards that push to the Report detail.
// Ported from pwa/src/screens/Projects.tsx (ProjectsList).
import SwiftUI

struct ProjectsScreen: View {
    @State private var filter: ProjectFilter = .all

    private var list: [Project] { DemoData.projects.filter { filter.matches($0) } }

    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(spacing: 0) {
                    portfolioCard
                        .padding(.horizontal, 20)
                        .padding(.top, 2)
                        .padding(.bottom, 8)

                    FilterChips(
                        items: ProjectFilter.allCases,
                        label: { $0.rawValue },
                        active: $filter
                    )
                    .padding(.bottom, 6)

                    LazyVStack(spacing: 10) {
                        ForEach(list) { project in
                            NavigationLink {
                                ReportDetailScreen(project: project)
                            } label: {
                                ProjectCard(project: project)
                            }
                            .buttonStyle(.plain)
                        }
                        if list.isEmpty {
                            Text("No projects in this filter.")
                                .font(.system(size: 14))
                                .foregroundStyle(T.faint)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 40)
                        }
                    }
                    .padding(.horizontal, 20)
                    .padding(.top, 4)
                    .padding(.bottom, 12)
                }
            }
            .background(T.bg.ignoresSafeArea())
            .navigationTitle("Projects")
            .navigationBarTitleDisplayMode(.large)
            .toolbar {
                ToolbarItem(placement: .topBarTrailing) {
                    Text("\(DemoData.projects.count) projects · Scotland")
                        .font(.system(size: 13, weight: .medium))
                        .foregroundStyle(T.muted)
                }
            }
        }
    }

    // Portfolio summary — the teal hero: avg ring + On track / Review / Complete counts.
    private var portfolioCard: some View {
        HStack(spacing: 16) {
            Ring(value: DemoData.portfolioAvg, size: 66, stroke: 6, accent: true, label: "AVG")
            HStack {
                statCount("On track", DemoData.onTrackCount, T.accent)
                Spacer()
                statCount("Review", DemoData.reviewCount, T.warning)
                Spacer()
                statCount("Complete", DemoData.completeCount, T.muted)
            }
        }
        .padding(16)
        .cardSurface()
    }

    private func statCount(_ label: String, _ n: Int, _ color: Color) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            Text("\(n)")
                .font(.system(size: 22, weight: .heavy))
                .foregroundStyle(color)
            Text(label)
                .font(.system(size: 11.5, weight: .semibold))
                .foregroundStyle(T.muted)
        }
    }
}

// MARK: - Project card

struct ProjectCard: View {
    let project: Project

    var body: some View {
        HStack(spacing: 14) {
            Ring(value: project.pct, size: 54)
            VStack(alignment: .leading, spacing: 3) {
                Text(project.name)
                    .font(.system(size: 16, weight: .bold))
                    .tracking(-0.3)
                    .foregroundStyle(T.text)
                    .lineLimit(1)
                Text("\(project.location) · \(project.area) m² · \(project.scans) scans")
                    .font(.system(size: 12.5))
                    .foregroundStyle(T.muted)
                    .lineLimit(1)
                StatusBadge(status: project.status, small: true)
                    .padding(.top, 5)
            }
            Spacer(minLength: 4)
            Image(systemName: "chevron.right")
                .font(.system(size: 14, weight: .semibold))
                .foregroundStyle(Color.white.opacity(0.22))
        }
        .padding(13)
        .frame(maxWidth: .infinity, alignment: .leading)
        .cardSurface()
    }
}
